pkg/client/client.go

```diff
@@ -215,11 +275,37 @@ func NewValidatedClient(ctrlRuntimeClient client.Client, secretName, namespace,
 	}
 
 	// Check that the endpoint can be resolved by the endpoint resolver.
+	// If the endpoint is not resolvable locally, we try to validate using the AWS API.
 	// If the endpoint is not known, it is not a standard or configured custom region.
 	// If this is the case, the client will likely not be able to connect
 	_, err = s.Config.EndpointResolver.EndpointFor("ec2", region, func(opts *endpoints.Options) {
 		opts.StrictMatching = true
 	})
+	if err != nil {
+		switch err.(type) {
+		case endpoints.UnknownEndpointError:
+			// Check again without strict matching to make sure region passes regex check
+			_, err = s.Config.EndpointResolver.EndpointFor("ec2", region, func(o *endpoints.Options) {
+				o.StrictMatching = false
+			})
+			if err != nil {
+				return nil, fmt.Errorf("invalid region %s: %v", region, err)
+			}
+
+			klog.Info("Region ", region, " is not recognized by aws-sdk, trying to validate using API")
```

Comment:
You should use `Infof` instead
```suggestion
			klog.Infof("Region %s is not recognized by aws-sdk, trying to validate using API", region)
```